### ISSUE TITLE
LR scheduling with Nvidia/APEX

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#545]
Probably, since https://github.com/NVIDIA/apex/pull/310 optimizer's param group is replaced and ignite's contrib lr schedulling is not applied to the actual param group.

```python
lr_scheduler = PiecewiseLinear(optimizer, param_name='lr', milestones_values=milestones_values)
len(optimizer.param_groups), id(optimizer.param_groups), id(optimizer.param_groups[0])
> (1, 139701776927624, 139701613652440)

from apex import amp
model, optimizer = amp.initialize(model, optimizer, opt_level="O2", num_losses=1)

len(optimizer.param_groups), id(optimizer.param_groups), id(optimizer.param_groups[0])
> (1, 139701467050824, 139701465940400)

id(lr_scheduler.optimizer_param_groups)
> 139701776927624
```

**Comment by vfdev-5** [comment]
If using opt_level="O1", ignite's params schedulers seem to work

**Comment by vfdev-5** [comment]
Fixed by #690
